internal/discovery/foreign-cluster-operator/auth.go

```diff
@@ -161,52 +187,103 @@ func (r *ForeignClusterReconciler) getAuthToken(fc *discoveryv1alpha1.ForeignClu
 	return ""
 }
 
-// send HTTP request to get the identity from the remote cluster.
-func (r *ForeignClusterReconciler) askRemoteIdentity(fc *discoveryv1alpha1.ForeignCluster) (string, error) {
+// askRemoteIdentity sends an HTTP request to get the identity from the remote cluster (ServiceAccount).
+func (r *ForeignClusterReconciler) askRemoteIdentity(fc *discoveryv1alpha1.ForeignCluster) ([]byte, error) {
 	token := r.getAuthToken(fc)
 
 	roleRequest := auth.ServiceAccountIdentityRequest{
 		ClusterID: r.clusterID.GetClusterID(),
 		Token:     token,
 	}
-	jsonRequest, err := json.Marshal(roleRequest)
+	return sendIdentityRequest(&roleRequest, fc)
+}
+
+// validateIdentity sends an HTTP request to validate the identity for the remote cluster (Certificate).
+func (r *ForeignClusterReconciler) validateIdentity(fc *discoveryv1alpha1.ForeignCluster) error {
+	remoteClusterID := fc.Spec.ClusterIdentity.ClusterID
+	token := r.getAuthToken(fc)
+
+	_, err := r.identityManager.CreateIdentity(remoteClusterID)
```

Comment:
What if the identity already exists?

Reply:
* If the identity already exists in the local cluster, this function will not be even invoked. This will only be invoked if there is not a secret containing the issued certificate.

* If the identity already exists in the remote cluster, it will return a forbidden error, because of the impossibility to validate a second identity for the same clusterID. If we accept the creation of a second identity with the same clusterID, when a cluster A is peered with a cluster B, and a cluster C knows the ID of cluster A, it will be able to forge an identity and to act as A taking control of its resources.

Reply:
Clear. Thanks!